file-manager/fileManager.cpp

```diff
@@ -0,0 +1,168 @@
+        std::cout << "Failed to open the file: " << filePath << std::endl;
+        return false;
+    }
+
+    // Get the file size.
+    std::streamsize fileSize = file.tellg();
+    file.seekg(0, std::ios::beg);
+
+    // Read the file content into a buffer
+    std::vector<char> buffer(fileSize);
+    if (!file.read(buffer.data(), fileSize)) 
+    {
+        std::cout << "Failed to read the file: " << filePath << std::endl;
+        return false;
+    }
+
+    // Create the binary object for bsoncxx.
+    bsoncxx::types::b_binary data{bsoncxx::binary_sub_type::k_binary, static_cast<std::uint32_t>(fileSize), reinterpret_cast<const std::uint8_t*>(buffer.data())};
+
+    // Create a document with the file name and file content.
+    auto doc = bsoncxx::builder::stream::document{};
+    doc << FILE_NAME << std::filesystem::path(filePath).filename();
+    doc << FILE_PATH << filePath;
+    doc << FILE_DATA << data;
+
+    // Insert the document into the collection.
+    collection.insert_one(doc.view());
+    file.close();
+
+    std::cout << "Upload successful for: " << filePath << std::endl;
+
+    return true;
+}
+
+// Download a file from a collection to a given folder.
+bool download(const std::string& fileName, const std::string& destiationFolder, mongocxx::collection& collection)
+{
+    // Create a query to find the file by filename
+    bsoncxx::builder::stream::document filter;
+    filter << FILE_NAME << fileName;
+    auto filterDoc = filter << bsoncxx::builder::stream::finalize;
+
+     // Find the document in the collection
+    auto result = collection.find_one(filterDoc.view());
+
+    if (result) 
+    {
+        // Get the binary data from the document
+        bsoncxx::document::view binaryDocView = result->view();
+        auto binaryData = binaryDocView[FILE_DATA].get_binary();
+        
+        // Create a file to save the binary data
+        std::ofstream file(destiationFolder + fileName, std::ios::binary);
+        if (!file) 
+        {
+            std::cout << "Failed to create the file: " << fileName << " at " << destiationFolder << std::endl;
+            return false;
+        }
+        
+        // Write the binary data to the file
+        file.write(reinterpret_cast<const char*>(binaryData.bytes), binaryData.size);
+        file.close();
+
+        std::cout << "Download sucessful for: " << fileName << " at " << destiationFolder << std::endl;
+        return true;
+    } 
+    else 
+    {
+        std::cout << "File not found in the collection: " << fileName << std::endl;
+        return false;
+    }
+
+}
+
+
+// ********************************************** Main **********************************************
+int main()
+{
+    try
+    {
+        auto mongoURIStr = "<Insert MongoDB Connection String>";
+        static const mongocxx::uri mongoURI = mongocxx::uri{ mongoURIStr };
+
+        if(mongoURI.to_string().empty())
+        {
+            std::cout << "URI is empty";
+            return 0;
+        }
```

Comment:
```suggestion
```

Suggest removing unnecessary check. If `mongoURIStr` is empty, this will result in an exception:

```
Exception encountered: Invalid URI Schema, expecting 'mongodb://' or 'mongodb+srv://': an invalid MongoDB URI was provided
```